src/main/java/ch/njol/skript/lang/function/FunctionRegistry.java

```diff
@@ -59,31 +66,33 @@ private FunctionRegistry() {
 	 * @throws SkriptAPIException if a signature with the same name and parameters is already registered
 	 *                            in this namespace.
 	 */
-	public static void register(@Nullable String namespace, @NotNull Signature<?> signature) {
-		Preconditions.checkNotNull(signature, "signature is null");
+	public synchronized void register(@Nullable String namespace, @NotNull Signature<?> signature) {
+		Preconditions.checkNotNull(signature, "signature cannot be null");
 		Skript.debug("Registering signature '" + signature.getName() + "'");
 
 		// namespace
-		Namespace ns = GLOBAL_NAMESPACE;
+		NamespaceIdentifier namespaceId = GLOBAL_NAMESPACE;
 		if (namespace != null && signature.isLocal()) {
-			ns = new Namespace(Scope.LOCAL, namespace);
+			namespaceId = new NamespaceIdentifier(Scope.LOCAL, namespace);
 		}
 
-		FunctionIdentifier identifier = FunctionIdentifier.of(signature);
+		synchronized (lock) {
```

Comment:
any reason why this doesn't synchronize on namespaces? Also why make the whole method synchronized?

Reply:
oops, forgot to remove the whole method synchronization. also forgot that you don't have to create objects for synchronization like you have to in c lol